src/domain/common.js

```diff
@@ -3,3 +3,14 @@ export function get1InchSwapUrl(chainId, from, to) {
   if (!from && !to) return rootUrl;
   return `${rootUrl}/${from}/${to}`;
 }
+
+export function getRootUrl() {
+  if (!window.location.origin) {
+    window.location.origin =
```

Comment:
Hm, does window.location.origin need to be mutated here?
Maybe just return a value? like const url = ...; return url

Also, seems more like 'lib' function than 'domain', i think it could be lib/url.ts 


It's not very obvious right now, but I hope in the future it will be something like this:

**domain** folder - this is logic specific to GMX, such as tokens, orders, positions, etc.

**lib** is low-level technical and infrastructure code. 

Your function `getRootUrl` could be in any other web project or from npm, then it goes in the 'lib' folder, you can also treat it as the 'utils' folder.

Why we separate these layers:
1. it's easier to find specific code without delving into the codebase, e.g:
1.1 I need some specific query for the GMX contract, for example for token statistics, and I want to see some data types and how they are organized - this should be somewhere in 'domain'.
1.2 I need some kind of url formatter or basic callContract function to implement the new query - then it's in 'lib'.

2 It helps to keep the code clear and control dependencies. Ideally the 'lib' code should be unaware of GMX, and remain flexible and reusable, the 'domain' code should contain clear business rules and logic around our data structures without a lot of low-level code about the environment (e.g. browser), frameworks, etc.

Reply:
Got it. Thanks for taking time to explain. 🙏